src/main/java/task/Task.java

```diff
@@ -0,0 +1,53 @@
+package task;
+
+/**
+ * Represents a task with a description and a completion status.
+ */
+public abstract class Task {
+    protected final String description;
+    private boolean done;
+
+    /**
+     * Constructs a new task.Task with the given description.
+     *
+     * @param description The description of the task.
+     */
+    public Task(String description) {
+        this.description = description.trim();
+        this.done = false;
```

Comment:
Try to reduce the use of `this`. Use of `this`​​ is discouraged, potentially to avoid unnecessary noise in the code. Try to avoid using this​​ except when it is shadowed by a method or constructor parameter.